// tera-auth-ticket/index.js
const webNA = require('./regions/na');
const webEU = require('./regions/eu');
const webRU = require('./regions/ru');

class webClient {
    constructor(region, email, password, account, interfaceAddress) {
        this.region = region;
        this.email = email;
        this.password = password;
        this.account = account;
        this.interfaceAddress = interfaceAddress;
        this.web;
    }
    async getLogin(callback){
        try {
            switch (this.region){
                case 'na': 
                    this.web = new webNA(this.email, this.password, this.interfaceAddress);
                    break;
                case 'eu': 
                    this.web = new webEU(this.email, this.password, this.account, this.interfaceAddress);
                    break;
                case 'ru':
                    this.web = new webRU(this.email, this.password, this.interfaceAddress);
                    break;
                default:
                    throw `Unsupported Region: "${this.region}"!`
            }
            const data = await this.web.getLogin();
            if(data && data.name && data.ticket) {
            	callback(null, data);
            } else {
            	callback('Failed to fetch auth ticket');
            }
        } catch(err){
            callback(err)
        }
    }
}

module.exports = webClient;
